src/Types/NullableType.hack

```diff
@@ -0,0 +1,64 @@
+namespace Slack\GraphQL\Types;
+
+use namespace Slack\GraphQL\Introspection;
+
+interface INullableType {
+    require extends BaseType;
+    public function getInnerType(): INonNullableType;
+}
+
+trait TNullableType implements INullableType {
+
+    /**
+     * GraphQL introspection doesn't use Nullable wrappers, so present this as an unwrapped type (delegate everything
+     * to the inner type).
+     */
+    <<__Override>>
+    final public function getIntrospectionKind(): Introspection\__TypeKind {
+        return $this->getInnerType()->getKind();
```

Comment:
we'd update these to `getIntrospectionKind`?

Reply:
Oh nvm. Read through the other one.

Reply:
No, this is intentional (this is the confusing part). When you introspect a type through a nullable wrapper, we return the *actual kind* of the inner type.

When you introspect the inner type directly, we pretend it's a non-null wrapper (getIntrospectionKind returns NON_NULL, getKind is not used at that point)